onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -181,37 +186,48 @@ common::Status SaveMLValueNameIndexMapping(const GraphViewer& graph_viewer,
   return Status::OK();
 }
 
-common::Status DeserializeTensorProto(const ONNX_NAMESPACE::TensorProto& tensor_proto,
-                                      const OrtAllocatorInfo& alloc_info,
-                                      const ExecutionProviders& exec_providers,
-                                      MLValue& mlvalue, void* preallocated, size_t preallocated_size) {
-  auto alloc_ptr = utils::GetAllocator(exec_providers, alloc_info);
-  if (!alloc_ptr) {
-    return Status(common::ONNXRUNTIME, common::FAIL, "Failed to get allocator for alloc_info: " + alloc_info.ToString());
-  }
-
+// This function won't check if the preallocated buffer(p_data) has enough room.
+common::Status DeserializeTensorProto(const Env& env, const std::basic_string<PATH_CHAR_TYPE>& proto_path,
+                                      const ONNX_NAMESPACE::TensorProto& tensor_proto, const MemBuffer& m,
+                                      const ExecutionProviders& exec_providers, MLValue& mlvalue, OrtDeleter& deleter) {
+  const OrtAllocatorInfo& alloc_info = m.GetAllocInfo();
   if (strcmp(alloc_info.name, CPU) == 0 || alloc_info.mem_type == OrtMemTypeCPUOutput) {
     // deserialize directly to CPU tensor
-    return utils::TensorProtoToMLValue(tensor_proto, alloc_ptr, preallocated, preallocated_size, mlvalue);
+    return utils::TensorProtoToMLValue(env, proto_path.c_str(), tensor_proto, m, mlvalue, deleter);
```

Comment:
>proto_path.c_str(), [](start = 44, length = 19)

If we're just calling c_str to pass proto_path, can we use a const std::string& for the param?